text.py

```diff
@@ -418,3 +418,93 @@ def result(self, state, action):
     def goal_test(self, state):
         """We're done when all letters in search domain are assigned."""
         return len(state) >= len(self.decoder.chardomain)
+#-----------------------------TF-IDF--------------------------------------
+class tf_idf_analysis:
+    '''a class to perform tf-idf analysis on a given set of documents and also search a query'''
+#     class variabels(their type)= values contained in the variable
+#     docs(list)=contains all the documents
+#     terms_tf_idf_score(list of dict)=tf-idf-score of all the documents with all words
+#     doc_tf(list of dict)=a list containting the tf of each word in each document
+#     terms_df(dict of list)= gives the df of ach term
+    def __init__(self,docs):
+        '''input: list of all documents'''
+        self.docs=docs
+        self.tf_idf_score=self.make_tf_idf()
+    def make_tf_idf(self):
+        '''makes the tf-idf score of all the words in all the documents'''
+        terms_df={}
+        doc_tf=[]
+        counter=0
+        for i in self.docs:
+            counter+=1
```

Comment:
In general, you need to add more space to your code to make it more readable. Add spaces around all operators, arguments and add empty lines every once in a while (around loops, for example), to break the code up a bit. Right now it looks like a huge wall of text.

This applies to all the code below this.

Reply:
@MrDupin Thank You for your detailed review of my code. I am working on the changes asked by you and update the PR soon. Thank You!

Reply:
Awesome, thanks!